### ISSUE TITLE
optimize_for_inference can't remove the nodes

### ISSUE THREAD

**Issue by civilman628** [tensorflow/tensorflow#5704]
i just try to train a new inception v2 slim model and freeze . ckpt into .pb file and then run optimize_for_inference to remove the nodes which are not used for inference.

firstly, i output all tensor names after get .pb file, before run **optimize_for_inference**.

- prefetch_queue/fifo_queue
- fifo_queue_Dequeue
- InceptionV2/Conv2d_1a_7x7/depthwise_weights
- InceptionV2/Conv2d_1a_7x7/pointwise_weights
- ...
- ...
- InceptionV2/Logits/AvgPool_1a_7x7/AvgPool
- InceptionV2/Logits/Conv2d_1c_1x1/weights
- InceptionV2/Logits/Conv2d_1c_1x1/Conv2D
- InceptionV2/Logits/Conv2d_1c_1x1/biases
- InceptionV2/Logits/Conv2d_1c_1x1/BiasAdd
- InceptionV2/Logits/SpatialSqueeze
- InceptionV2/Predictions/Reshape/shape
- InceptionV2/Predictions/Reshape
- InceptionV2/Predictions/Softmax

===================

for test purpose, if i assign 
**--input_names=fifo_queue_Dequeue**   // it is the 2nd node (line 2 above) of .pb file
**--output_names=InceptionV2/Predictions/Softmax**
After run **optimize_for_inference**, the first node **prefetch_queue/fifo_queue** (line 1) is still there. It does not by pass any node for the optimized .pb file.

My goal is input a 4d image tensor [batch size, W, H, channel] or a 4d numpy array inside, but i cannot find where to insert.

**Comment by civilmanxx** [comment]
such as get .pb file and predict a single image by using **classify_image.py**
